src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -178,5 +182,38 @@ public bool ManualSettingsOnly
 
             return useProxy;
         }
+
+        public bool AutoDetectScriptSuccess()
+        {
+            // Only check WinHttpDetectAutoProxyConfigUrl if auto detect is set.
+            if (!AutoDetect) return false;
+
+            if (_autoDetectTimer == null)
+            {
+                _autoDetectTimer = new Timer(new TimerCallback(UpdateScriptDetectionStatus));
```

Comment:
I don't like the idea of Timer. I think it has higher overhead than just checking timestamp as we discussed offline.
Also, it is not pay-for-play as it will trigger on regular basis even if there are not any requests.

Reply:
Will fix.